api/src/main/java/net/hypejet/jet/plugin/PluginDependency.java

```diff
@@ -0,0 +1,14 @@
+package net.hypejet.jet.plugin;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Represents a dependency of a plugin.
+ *
+ * @param pluginName a name of the plugin, which is the dependency
+ * @param required whether the plugin is required
+ * @since 1.0
+ * @author Codestech
+ * @see PluginMetadata
+ */
+public record PluginDependency(@NonNull String pluginName, boolean required) {}
```

Comment:
this should also have a version property (should support both =, >= and <=), like every other plugin / mod loader

Reply:
What do you mean by `=`, `>=` and `<=`? Like the plugin can run on a version specified (`=`), version higher or the same as specified (`>=`) and version lower or the same as specified (`<=`)?

Reply:
yup